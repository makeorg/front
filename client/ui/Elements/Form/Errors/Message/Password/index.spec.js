import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import { ErrorMessageForgotPasswordComponent } from './index';

describe('Error Message Forgot Password', () => {
  it('snapshot ErrorMessageForgotPasswordComponent', () => {
    const component = renderer
      .create(
        <ErrorMessageForgotPasswordComponent
          handleForgotPasswordModal={() => {}}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('snapshot ErrorMessageForgotPasswordComponent default vs custom label', () => {
    const RequiredTextArea = renderer
      .create(
        <ErrorMessageForgotPasswordComponent
          handleForgotPasswordModal={() => {}}
        />
      )
      .toJSON();
    const OptionnalTextArea = renderer
      .create(
        <ErrorMessageForgotPasswordComponent
          handleForgotPasswordModal={() => {}}
          inputId="foo"
          labelText="fooLabelText"
        />
      )
      .toJSON();
    expect(snapshotDiff(RequiredTextArea, OptionnalTextArea)).toMatchSnapshot();
  });
});
