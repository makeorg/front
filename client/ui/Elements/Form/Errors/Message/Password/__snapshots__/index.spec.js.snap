// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message Forgot Password snapshot ErrorMessageForgotPassword 1`] = `
Array [
  <span
    dangerouslySetInnerHTML={
      Object {
        "__html": "profile.password_update.actual_password.invalid_password",
      }
    }
  />,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  border: none;
  background-color: transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: 'Circular Standard Bold',Arial,sans-serif;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-left: 5px;
}

<button
    className="sc-11o2adg-1 c0"
    onClick={[Function]}
  >
    profile.password_update.actual_password.trigger
  </button>,
]
`;

exports[`Error Message Forgot Password snapshot ErrorMessageForgotPassword default vs custom label 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
